manager/controllers/app/moduleinstance.go

```diff
@@ -25,8 +25,23 @@ type ModuleManager struct {
 }
 
 // SelectModuleInstances builds a list of required modules with the relevant arguments
-/* The order of the lookup is Read, Copy, Write.
-   Assumptions:
+/*
+
+Future (ingest & write) order of the lookup to support ingest is:
+- If no label selector assume ingest of external data (what about archive in future?)
+	- run Copy module close to destination (determined based on governance decisions)
+	- and register new data set in data catalog
+- If Data Context Flow=Write
+   - Write is always required, and always close to compute
+   - Implicit Copy is used on demand, e.g. if a write module does not support the existing source of data or governance actions
+   - Transformations are always done at workload location
+   - If not external data, then register in data catalog
+
+Updates to add ingest:
+- If no label selector assume ingest of external data
+	- run Copy module close to destination (determined based on governance decisions)
```

Comment:
I don't see how this is expressed in the code.

Reply:
The copy module SelectCluster determines where it will run.  No changes were required to it.  I sets the cluster to the destination if no transforms are required, which is fine.